php/ext/google/protobuf2/php-upb.h

```diff
@@ -0,0 +1,3885 @@
+/* Amalgamated source file */
+#include <stdint.h>/*
+* This is where we define macros used across upb.
+*
+* All of these macros are undef'd in port_undef.inc to avoid leaking them to
+* users.
+*
+* The correct usage is:
+*
+*   #include "upb/foobar.h"
+*   #include "upb/baz.h"
+*
+*   // MUST be last included header.
+*   #include "upb/port_def.inc"
+*
+*   // Code for this file.
+*   // <...>
+*
+*   // Can be omitted for .c files, required for .h.
+*   #include "upb/port_undef.inc"
+*
+* This file is private and must not be included by users!
+*/
+#include <stdint.h>
+#include <stddef.h>
+
+#if UINTPTR_MAX == 0xffffffff
```

Comment:
[Portability] It may be cleaner to check `sizeof(uintptr_t) == sizeof(uint32_t)`. Alternatively, it might be good to add `ul` to the constant.

Reply:
> It may be cleaner to check sizeof(uintptr_t) == sizeof(uint32_t).

That doesn't work because we need to resolve this in the preprocessor, and `sizeof()` isn't evaluated until later.

> Alternatively, it might be good to add ul to the constant.

I don't think this is necessary, the preprocessor will treat all integer constants as either `intmax_t` or `uintmax_t`, depending on whether they are signed or unsigned.

Reply:
As usual, I'm blind. :-)

I was mostly trying to preserve signedness to match the internals (and clarify there is no signed extension), e.g.:
https://github.com/llvm/llvm-project/blob/38b39c3/clang/test/Preprocessor/init.c#L6174
(which is also in clang -E -dM)

But this should be fine.